microservices/matchmaking-service/src/scrim/scrim-logic/scrim-logic.service.ts

```diff
@@ -25,24 +27,28 @@ export class ScrimLogicService {
         scrim.status = ScrimStatus.POPPED;
         scrim.submissionId = `scrim-${uuid()}`;
 
-        const job = await this.scrimQueue.add("timeoutQueue", scrim.id, {delay: scrim.settings.checkinTimeout});
+        if (scrim.players.some(p => p.canSaveDemos)) {
+            const job = await this.scrimQueue.add("timeoutQueue", scrim.id, {delay: scrim.settings.checkinTimeout});
 
-        await this.scrimCrudService.updateScrimStatus(scrim.id, scrim.status);
-        await this.scrimCrudService.setSubmissionId(scrim.id, scrim.submissionId);
-        await this.scrimCrudService.setTimeoutJobId(scrim.id, job.id);
+            await this.scrimCrudService.updateScrimStatus(scrim.id, scrim.status);
+            await this.scrimCrudService.setSubmissionId(scrim.id, scrim.submissionId);
+            await this.scrimCrudService.setTimeoutJobId(scrim.id, job.id);
 
-        const updatedScrim = await this.scrimCrudService.getScrim(scrim.id);
-        if (!updatedScrim) throw new Error("Scrim is somehow missing!");
-        await this.eventsService.publish(EventTopic.ScrimPopped, updatedScrim, scrim.id);
+            const updatedScrim = await this.scrimCrudService.getScrim(scrim.id);
+            if (!updatedScrim) throw new Error("Scrim is somehow missing!");
+            await this.eventsService.publish(EventTopic.ScrimPopped, updatedScrim, scrim.id);
 
-        this.analyticsService
-            .send(AnalyticsEndpoint.Analytics, {
-                name: "scrimPopped",
-                strings: [["scrimId", scrim.id]],
-            })
-            .catch(err => {
-                this.logger.error(err);
-            });
+            this.analyticsService
+                .send(AnalyticsEndpoint.Analytics, {
+                    name: "scrimPopped",
+                    strings: [["scrimId", scrim.id]],
+                })
+                .catch(err => {
+                    this.logger.error(err);
+                });
+        } else {
+            await this.scrimService.cancelScrim(scrim.id);
```

Comment:
We should probably do something other than blindly cancelling. Can we send a notification to players saying why the scrim was cancelled?

Reply:
Not through UI unless we add a "cancelled reason" to the scrim cancelled event or if we just do a Discord DM, which we should do but that shouldn't be THE source.